File: Software/BrickPi_C/Sensor_Examples/LEGO - ColorSensor test.c
/*
*  LEGO Color Sensor Test.
*
*  t.s.jaikrishna<at>gmail.com
*  Initial date: June 20, 2013
*  Updated:  Feb 17, 2015 (John)
*  Based on Matthew Richardson's Example for testing BrickPi
*  You may use this code as you wish, provided you give credit where it's due.
*
*  This is a program for testing the RPi BrickPi driver with Lego Color Sensor on 4th Port
*/

#include <stdio.h>
#include <math.h>
#include <time.h>

#include "tick.h"
#include "BrickPi.h"

#include <linux/i2c-dev.h>  
#include <fcntl.h>

// Compile Using:
// sudo gcc -o program "LEGO - ColorSensor test.c" -lrt -lm -L/usr/local/lib -lwiringPi
// Run the compiled program using:
// sudo ./program

int result;
#undef DEBUG


int main() {
  ClearTick();
  char* col[] = {"t", "Black","Blue","Green","Yellow","Red","White"};
  result = BrickPiSetup();
  printf("BrickPiSetup: %d\n", result);
  if(result)
    return 0;

  BrickPi.Address[0] = 1;
  BrickPi.Address[1] = 2;

  BrickPi.SensorType[PORT_4] = TYPE_SENSOR_COLOR_FULL;

  result = BrickPiSetupSensors();
  printf("BrickPiSetupSensors: %d\n", result); 
  if(!result){
    
    usleep(10000);
    
    while(1){
      result = BrickPiUpdateValues();

      if(!result){
      	
         printf("Results: %d %s\n", BrickPi.Sensor[PORT_4], col[BrickPi.Sensor[PORT_4]]);  //BrickPi.Sensor returns a number from 1 to 6. The col array contains the corresponding color names
   
       }
      usleep(10000);
    }
  }
  return 0;
}

File: Software/BrickPi_Python/Sensor_Examples/LEGO-Color_Sensor_Test.py
#!/usr/bin/env python
# Jaikrishna
# Initial Date: June 24, 2013
# Updated:      Oct 21, 2014 by John
# Updated:      Oct 28, 2016 by Shoban
#
# These files have been made available online through a Creative Commons Attribution-ShareAlike 3.0  license.
# (http://creativecommons.org/licenses/by-sa/3.0/)
#
# This code is for testing the BrickPi with a Lego Color Sensor
# Connect the LEGO Color sensor to Sensor port S1.
#
# You can learn more about BrickPi here:  http://www.dexterindustries.com/BrickPi
# Have a question about this example?  Ask on the forums here:  http://forum.dexterindustries.com/c/brickpi

#For the code to work - sudo pip install -U future

from __future__ import print_function
from __future__ import division
from builtins import input

# the above lines are meant for Python3 compatibility.
# they force the use of Python3 functionality for print(), 
# the integer division and input()
# mind your parentheses!

from BrickPi import *   #import BrickPi.py file to use BrickPi operations

BrickPiSetup()  # setup the serial port for communication

Color_Sensor_Port = PORT_1										# Setup the sensor on Port 1.
col = [ None , "Black","Blue","Green","Yellow","Red","White" ]   #used for converting the color index to name
BrickPi.SensorType[Color_Sensor_Port] = TYPE_SENSOR_COLOR_FULL   #Set the type of sensor 


BrickPiSetupSensors()   #Send the properties of sensors to BrickPi

while True:
    result = BrickPiUpdateValues()  # Ask BrickPi to update values for sensors/motors 
    if not result :
        print(col[BrickPi.Sensor[Color_Sensor_Port]])     #BrickPi.Sensor[PORT] stores the value obtained from sensor
    time.sleep(.1)     # sleep for 100 ms
